scripts/west_commands/run_common.py

```diff
@@ -142,90 +142,97 @@ def do_run_common(command, user_args, user_runner_args):
-    # override the general:
-    #
-    # - common runners.yaml arguments
-    # - runner-specific runners.yaml arguments
-    # - command line arguments
-    final_argv = (runner_config['args']['common'] +
-                  runner_config['args'][runner_name] +
-                  runner_args)
-
-    # 'user_args' contains parsed arguments which are:
-    #
-    # 1. provided on the command line, and
-    # 2. handled by add_parser_common(), and
-    # 3. *not* runner-specific
-    #
-    # 'final_argv' contains unparsed arguments from either:
-    #
-    # 1. runners.yaml, or
-    # 2. the command line
-    #
-    # We next have to:
-    #
-    # - parse 'final_argv' now that we have all the command line
-    #   arguments
-    # - create a RunnerConfig using 'user_args' and the result
-    #   of parsing 'final_argv'
-    parser = argparse.ArgumentParser(prog=runner_name)
-    add_parser_common(command, parser=parser)
-    runner_cls.add_parser(parser)
-    args, unknown = parser.parse_known_args(args=final_argv)
-    if unknown:
-        log.die(f'runner {runner_name} received unknown arguments: {unknown}')
-
-    # Override args with any user_args. The latter must take
-    # precedence, or e.g. --hex-file on the command line would be
-    # ignored in favor of a board.cmake setting.
-    for a, v in vars(user_args).items():
-        if v is not None:
-            setattr(args, a, v)
-
-    # Create the RunnerConfig from the values assigned to common
-    # arguments. This is a hacky way to go about this; probably
-    # ZephyrBinaryRunner should define what it needs to make this
-    # happen by itself. That would be a larger refactoring of the
-    # runners package than there's time for right now, though.
-    #
-    # Use that RunnerConfig to create the ZephyrBinaryRunner instance
-    # and call its run().
-    runner = runner_cls.create(runner_cfg_from_args(args, build_dir), args)
-    try:
-        runner.run(command_name)
-    except ValueError as ve:
-        log.err(str(ve), fatal=True)
-        dump_traceback()
-        raise CommandError(1)
-    except MissingProgram as e:
-        log.die('required program', e.filename,
-                'not found; install it or add its location to PATH')
-    except RuntimeError as re:
-        if not user_args.verbose:
-            log.die(re)
-        else:
-            log.err('verbose mode enabled, dumping stack:', fatal=True)
-            raise
+    for current_yaml in runners_yaml:
+        # A CMake list may contain multiple `;;` which can result in an empty
+        # entry in the array, thus simply ignore empty elements
+        if not current_yaml:
+            continue
+        runner_config = load_runners_yaml(current_yaml, user_args)
+
+        # Get a concrete ZephyrBinaryRunner subclass to use based on
+        # runners.yaml and command line arguments.
+        runner_cls = use_runner_cls(command, board, user_args, runner_config,
```

Comment:
Testing this a bit more, even in the single-yaml-file case, I think it makes sense to pass use_runner_cls() the current_yaml value, so it can be logged:

```
def use_runner_cls(command, board, args, runner_config, cache, yaml_file):
...
    _banner(f'west {command.name}: using runner {runner} ({yaml_file})')
```

Reply:
good point and I agree.
Will add dedicated commit for that improvement.